src/tree/AttachedServersTreeItem.ts

```diff
@@ -105,17 +132,97 @@ export class AttachedServersTreeItem implements IAzureParentTreeItem {
     private async loadPersistedServers() {
         const value: any = this._globalState.get(this._serviceName);
         if (value && this._keytar) {
-            const ids: string[] = JSON.parse(value);
-            await Promise.all(ids.map(async id => {
+            const accounts: any[] = JSON.parse(value);
+            await Promise.all(accounts.map(async account => {
+                let id: string;
+                let api: Experience;
+                if (typeof (account) === 'string') {
+                    // Default to Mongo if the value is a string for the sake of backwards compatiblity
+                    // (Mongo was originally the only account type that could be attached)
+                    id = account;
+                    api = Experience.MongoDB;
+                } else {
+                    id = (<IPersistedAccount>account).id;
+                    api = (<IPersistedAccount>account).defaultExperience;
+                }
                 const connectionString: string = await this._keytar.getPassword(this._serviceName, id);
-                const treeItem = new MongoAccountTreeItem(id, id, connectionString, 'mongoServer');
-                this._attachedServers.push(treeItem);
+                this._attachedAccounts.push(await this.createTreeItem(connectionString, api, id));
             }));
         }
     }
 
+    private async createTreeItem(connectionString: string, api: Experience, id?: string): Promise<IAzureTreeItem> {
+        let treeItem: IAzureTreeItem;
+        if (api === Experience.MongoDB) {
+            if (id === undefined) {
+                id = await this.getServerIdFromConnectionString(connectionString);
+            }
+
+            treeItem = new MongoAccountTreeItem(id, id, connectionString);
+        } else {
+            const [endpoint, masterKey, id] = AttachedAccountsTreeItem.parseDocDBConnectionString(connectionString);
+
+            switch (api) {
+                case Experience.Table:
+                    treeItem = new TableAccountTreeItem(id, id, endpoint, masterKey);
+                    break;
+                case Experience.Graph:
+                    treeItem = new GraphAccountTreeItem(id, id, endpoint, masterKey);
+                    break;
+                case Experience.DocumentDB:
+                default:
```

Comment:
Wouldn't it be better to explicitly reference all handled cases and throw on default?